agent/doctor/docker_runtime_healthcheck.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//      http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package doctor
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	"github.com/cihub/seelog"
+)
+
+type dockerRuntimeHealthcheck struct {
+	mu sync.RWMutex
```

Comment:
remind me what uses this struct concurrently that we need a `Mutex`?

Reply:
@angelcar This is currently not used concurrently, this struct will be written by a healthcheck and read by a queue, such that there might be a race condition. To be added in coming PRs.

Reply:
there was a comment in a previous review requiring these mutexes.  Since there's only one doctor per agent, they're likely not going to be necessary but as previous comments noted -- it's better to err on the side of caution.

Reply:
I guess we will see if we end up using it... but when it comes to concurrency, avoiding the usage of mutexes is often a more cautious choice than using them when they're not really needed.